runtime/src/main/java/org/corfudb/runtime/view/ManagementView.java

```diff
@@ -320,125 +238,303 @@ private NodeStatus getLogUnitNodeStatusInLayout(Layout layout, String server) {
+        return new ClusterStatusReport(layout,
+                ClusterStatus.UNAVAILABLE,
+                reliability,
+                layoutServers.stream().collect(Collectors.toMap(
+                        endpoint -> endpoint,
+                        node -> NodeStatus.NA)),
+                layoutServers.stream().collect(Collectors.toMap(
+                        endpoint -> endpoint,
+                        endpoint -> responsiveServers.contains(endpoint) ? ConnectivityStatus.RESPONSIVE : ConnectivityStatus.UNRESPONSIVE)));
+    }
+
+    private ClusterStatusReport getUnavailableClusterStatusReport(List<String> layoutServers) {
+        return getUnavailableClusterStatusReport(layoutServers, ClusterStatusReliability.UNAVAILABLE, Collections.emptySet(), null);
+    }
 
-        List<String> layoutServers
-                = new ArrayList<>(runtime.getLayoutServers());
-        Layout layout = getHighestEpochLayout(layoutServers);
-
-        // If layout is null, none of the existing layout servers have a layout.
-        if (layout == null) {
-            return new ClusterStatusReport(null,
-                    layoutServers.stream().collect(Collectors.toMap(
-                            endpoint -> endpoint,
-                            node -> NodeStatus.DOWN)),
-                    ClusterStatus.UNAVAILABLE);
+    private Map<String, Layout> getLayouts(List<String> layoutServers, Boolean discoverLayoutServers) {
+        Map<String, CompletableFuture<Layout>> layoutFuturesMap = getLayoutFutures(layoutServers);
+        Map<String, Layout> layoutMap = new HashMap<>();
+
+        // Wait on the Completable futures
+        for (Entry<String, CompletableFuture<Layout>> entry : layoutFuturesMap.entrySet()) {
+            try {
+                Layout nodeLayout = entry.getValue().get();
+                log.debug("Server:{} responded with layout: {}", entry.getKey(), nodeLayout);
+                layoutMap.put(entry.getKey(), nodeLayout);
+            } catch (Exception e) {
+                log.error("getClusterStatus: Error while fetching layout from {}. Exception: {}",
+                        entry.getKey(), e);
+            }
         }
 
-        RuntimeLayout runtimeLayout = new RuntimeLayout(layout, runtime);
+        // If discoverLayoutServers is set, fetch layouts from all other layout servers
+        // discovered in the first search which are not contained in the initial list.
+        if (discoverLayoutServers) {
+            List<String> discoveredLayoutServers = layoutMap.values().stream()
+                    .map(l -> l.getLayoutServers())
+                    .flatMap(s -> s.stream())
+                    .distinct()
+                    .filter(s -> !layoutServers.contains(s))
+                    .collect(Collectors.toList());
+
+            if (!discoveredLayoutServers.isEmpty()) {
+                Map <String, Layout> recursiveLayouts = getLayouts(new ArrayList<>(discoveredLayoutServers), false);
```

Comment:
Can we use an enum instead of boolean?

Reply:
why? a enum with a unique value? Besides, remember this will change to read from quorum api.

Reply:
Boolean doesn't describe what is that, so, if it requires changing API, then nevermind

Reply:
I've added a description, however as I said this method is likely to  be replaced.